dev/lint-python

```diff
@@ -38,7 +38,7 @@ function compile_python_test {
 
     # compileall: https://docs.python.org/2/library/compileall.html
     echo "starting python compilation test..."
-    COMPILE_REPORT=$( (python -B -mcompileall -q -l $1) 2>&1)
+    COMPILE_REPORT=$( (python -B -mcompileall -q -l -x "[/\\][.]git" $1) 2>&1)
```

Comment:
Just curious. I thought we need to exclude the `.git` folder only.
Why do we need to match `[/\\]`

Reply:
Hi, @gengliangwang . It's required because we want to match pathes with prefix `/.git` or `\.git` (windows). Otherwise, the regex can be match with file names like `a.git.py`, it's a standard way.
```
$ python -B -mcompileall -q -l -x "[.]git" ./a.git.py
$ python -B -mcompileall -q -l -x "[/][.]git" ./a.git.py
*** Error compiling './a.git.py'...
  File "./a.git.py", line 38
    clasz VectorTests(MLlibTestCase):
                    ^
SyntaxError: invalid syntax
```

However, it seems that I missed the fact that bash escaping happens here.